src/webgpu/api/validation/copy_between_linear_data_and_texture/copyBetweenLinearDataAndTexture_textureRelated.spec.ts

```diff
@@ -0,0 +1,305 @@
+
+g.test('sample_count_must_be_1')
+  .params(
+    params()
+      .combine(poptions('method', kAllTestMethods))
+      .combine(poptions('sampleCount', [1, 4]))
+  )
+  .fn(async t => {
+    const { sampleCount, method } = t.params;
+
+    const texture = t.device.createTexture({
+      size: { width: 4, height: 4, depth: 1 },
+      sampleCount,
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST | GPUTextureUsage.SAMPLED,
+    });
+
+    const success = sampleCount === 1;
+
+    t.testRun(
+      { texture },
+      { bytesPerRow: 0 },
+      { width: 0, height: 0, depth: 0 },
+      { dataSize: 1, method, success }
+    );
+  });
+
+g.test('mip_level_must_be_in_range')
+  .params(
+    params()
+      .combine(poptions('method', kAllTestMethods))
+      .combine(poptions('mipLevelCount', [3, 5]))
+      .combine(poptions('mipLevel', [3, 4]))
+  )
+  .fn(async t => {
+    const { mipLevelCount, mipLevel, method } = t.params;
+
+    const texture = t.device.createTexture({
+      size: { width: 32, height: 32, depth: 1 },
+      mipLevelCount,
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.COPY_DST,
+    });
+
+    const success = mipLevel < mipLevelCount;
+
+    t.testRun(
+      { texture, mipLevel },
+      { bytesPerRow: 0 },
+      { width: 0, height: 0, depth: 0 },
+      { dataSize: 1, method, success }
+    );
+  });
+
+g.test('texel_block_alignments_on_origin')
+  .params(
+    params()
+      .combine(poptions('method', kAllTestMethods))
+      .combine(poptions('coordinateToTest', ['x', 'y', 'z'] as const))
+      .combine(poptions('format', kTextureFormats))
+      .filter(formatCopyableWithMethod)
+      .expand(texelBlockAlignmentTestExpanderForValueToCoordinate)
+  )
+  .fn(async t => {
+    const { valueToCoordinate, coordinateToTest, format, method } = t.params;
+
+    const origin = { x: 0, y: 0, z: 0 };
+    const size = { width: 0, height: 0, depth: 0 };
+    let success = true;
+
+    origin[coordinateToTest] = valueToCoordinate;
+    switch (coordinateToTest) {
+      case 'x': {
+        success = origin.x % kTextureFormatInfo[format].blockWidth! === 0;
+        break;
+      }
+      case 'y': {
+        success = origin.y % kTextureFormatInfo[format].blockHeight! === 0;
+        break;
+      }
+    }
+
+    const texture = t.createAlignedTexture(format, size, origin);
+
+    t.testRun({ texture, origin }, { bytesPerRow: 0 }, size, {
+      dataSize: 1,
+      method,
+      success,
+    });
+  });
+
+g.test('1d_texture')
+  .params(
+    params()
+      .combine(poptions('method', kAllTestMethods))
+      .combine(
+        poptions('size', [
+          { width: 1, height: 0, depth: 0 },
+          { width: 0, height: 0, depth: 0 },
+          { width: 0, height: 0, depth: 1 },
+          { width: 0, height: 1, depth: 0 },
+          { width: 0, height: 0, depth: 2 },
+          { width: 0, height: 2, depth: 0 },
+        ])
+      )
```

Comment:
Should test a few more cases here:
- height and depth are 1 (success should be true)
- height=1,depth=2 and height=2,depth=1, because height=0,depth=2 and height=2,depth=0 both trigger two errors each.
- may as well test width=0,1 with all cases

```suggestion
      .combine(poptions('width', [0, 1]))
      .combine([
        { height: 1, depth: 1 },
        { height: 1, depth: 0 },
        { height: 1, depth: 2 },
        { height: 0, depth: 1 },
        { height: 2, depth: 1 },
      ])
```